services/venice-server/src/main/java/com/linkedin/venice/listener/ServerStoreAclHandler.java

```diff
@@ -58,11 +59,15 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
   }
 
   /**
-   * In Venice Server, the resource name is actually a Kafka topic name.
+   * In Venice Server, the resource name is actually a Kafka topic name for STORAGE/COMPUTE but store name for DICTIONARY.
    */
   @Override
-  protected String extractStoreName(String resourceName) {
-    return Version.parseStoreFromKafkaTopicName(resourceName);
+  protected String extractStoreName(String resourceName, QueryAction queryAction) {
+    if (QueryAction.DICTIONARY.equals(queryAction)) {
+      return resourceName;
+    } else {
+      return Version.parseStoreFromKafkaTopicName(resourceName);
+    }
```

Comment:
Should we explicitly handle `STORAGE`, `COMPUTE`, `DICTIONARY` and fail fast for the rest? That way, we'd enforce that this stays updated as the code evolves and new QueryActions are added

Reply:
Sure, we will basically have a double safe to catch outdated code here. One is in the test I added and second will be this method throwing explicit error.